### PR TITLE
Feat/ve mento locking contract

### LINKED ISSUE #311: Update ingress-nginx controller

### PR DESCRIPTION
**Pull request by baroooo**
### Description

Staking mechanism where users can lock their MENTO. In return, they receive veMENTO which grant them governance rights depending on the amount they staked and the staking schedule.

We forked rari's locking-contracts implementation @4f189a96b3e85602dedfbaf69d9a1f5056d835eb

https://github.com/rarible/locking-contracts/tree/4f189a96b3e85602dedfbaf69d9a1f5056d835eb

Changes we introd:
- Block time related changes between two blockchains
- Removing redundant and old code that is not used
- Move tests to forge

### Other changes

N/A

### Tested

Yes

### Related issues

- Fixes #311

### Backwards compatibility

N/A

### Documentation

N/A

### CODE REVIEW

**Review by philbow61 — APPROVED** [review]
LGTM 🚢 
- I think it might make sense to restructure/refactor those migrated hardhat tests in the future. I feel like they are not really nice and some cases are missing.

File: contracts/governance/locking/libs/LibBrokenLine.sol [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    }
+    brokenLine.initial.start = toTime;
+    brokenLine.initial.bias = bias;
+    brokenLine.initial.slope = slope;
+  }
+
+  function actualValue(
+    BrokenLine storage brokenLine,
+    uint32 toTime,
+    uint32 toBlock
```
[the remark this answers is not in the document]
Comment (baroooo) [reply contracts/governance/locking/libs/LibBrokenLine.sol:207]: I think unused `toBlock` is in `actualValueForward()` but yeah we can remove that.

**Review by bayological — COMMENTED** [review]
Overall I think we could do better with adding comments in functions for the forked code. Right now there is no consistency with this, for example, LockingVotes has simple comments that explain the intention of functions whereas Locking, LockingBase & LockingRelock have none. Comments similar to those in LockingVotes would be valuable to future dev or anyone looking who is not familiar with the Broken Line lib.

**Review by bayological — APPROVED** [review]

**Comment by baroooo** [comment]
created an issue to add comments.

**Pull request merged**